### PR TITLE
Device manager - device list filtering (PSG-648)

### PR DESCRIPTION
**Pull request by kerryarchibald**
Screenshot 2022-08-12 at 14 17 11
Screenshot 2022-08-12 at 14 17 03
Screenshot 2022-08-12 at 14 16 50

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Notes: Device manager - device list filtering

---
Here's what your changelog entry will look like:

## ✨ Features
 * Device manager - device list filtering ([\#9181](https://github.com/matrix-org/matrix-react-sdk/pull/9181)). Contributed by @kerryarchibald.

### CODE REVIEW

File: test/components/views/settings/devices/FilteredDeviceList-test.tsx [hunk 1]
```diff
@@ -15,25 +15,39 @@ limitations under the License.
...
-import { render } from '@testing-library/react';
+import { act, fireEvent, render } from '@testing-library/react';
 
 import FilteredDeviceList from '../../../../../src/components/views/settings/devices/FilteredDeviceList';
+import { DeviceSecurityVariation } from '../../../../../src/components/views/settings/devices/types';
+import { flushPromises, mockPlatformPeg } from '../../../../test-utils';
 
+mockPlatformPeg();
+
+const MS_DAY = 86400000;
```
Comment (SimonBrandner) [inline test/components/views/settings/devices/FilteredDeviceList-test.tsx:26]: Could this please be expressed as multiplication To be readable?

File: src/components/views/settings/devices/filter.ts [hunk 2]
```diff
@@ -18,7 +18,9 @@ import { DeviceWithVerification, DeviceSecurityVariation } from "./types";
 
 type DeviceFilterCondition = (device: DeviceWithVerification) => boolean;
 
+const MS_DAY = 24 * 60 * 60 * 1000;
 export const INACTIVE_DEVICE_AGE_MS = 7.776e+9; // 90 days
```
Comment (SimonBrandner) [inline src/components/views/settings/devices/filter.ts:22]: This could be more readable

**Comment by weeman1337** [comment]
Can you have a look at the filter select? It doesn't match the designs

image

File: src/components/views/settings/devices/FilteredDeviceList.tsx [hunk 3]
```diff
@@ -16,40 +16,178 @@ limitations under the License.
...
+            return _t('No verified sessions found.');
+        case DeviceSecurityVariation.Unverified:
+            return _t('No unverified sessions found.');
+        case DeviceSecurityVariation.Inactive:
+            return _t('No inactive sessions found.');
+        default:
+            return _t('No sessions found.');
+    }
+};
+interface NoResultsProps { filter: DeviceSecurityVariation, clearFilter: () => void}
```
Comment (SimonBrandner) [inline src/components/views/settings/devices/FilteredDeviceList.tsx:106]: Could this be multiline, please? I think it would be more readable

**Comment by weeman1337** [comment]
Maybe it was like this before. The cards should have the same vertical spacing. Currently some `p` elements add a margin:

image

File: src/components/views/settings/devices/FilteredDeviceList.tsx [hunk 4]
```diff
@@ -16,40 +16,178 @@ limitations under the License.
...
+            id: DeviceSecurityVariation.Inactive,
+            label: _t('Inactive'),
+            description: _t(
+                'Inactive for %(inactiveAgeDays)s days or longer',
+                { inactiveAgeDays: INACTIVE_DEVICE_AGE_DAYS },
+            ),
+        },
+    ];
+
+    const onFilterOptionChange = (filterId: DeviceSecurityVariation | string) => {
```
Comment (SimonBrandner) [inline src/components/views/settings/devices/FilteredDeviceList.tsx:155]:
```suggestion
    const onFilterOptionChange = (filterId: DeviceSecurityVariation | ALL_FILTER_ID | string) => {
```

**Review by SimonBrandner — APPROVED** [review]
LGTM! Just a few readability nits

**Review by weeman1337 — COMMENTED** [review]
Looks good. Tested the filter on the deployment :+1: 

Found some design issues only.

**Comment by kerryarchibald** [comment]
@weeman1337 thanks for reviewing! Fixed the card margin. I'm just using the existing dropdown in this PR, will address adding the new designs separately (have an open task in jira for it)

**Review by weeman1337 — APPROVED** [review]

**Pull request merged**